docs/sc-memory/api/cpp/extended/agents/agents.md

```diff
@@ -0,0 +1,660 @@
+// class `action_calculate_set_power` and that it has argument.
+
+// Full result condition of agent
+..agent_calculate_set_power_result_condition
+= [*
+    action_calculate_set_power _-> .._action;;
+    action_initiated _-> .._action;;
+    action_finished_successfully _-> .._action;;
+    .._action _-> rrel_1:: .._parameter;;
+    .._action _=> nrel_result:: .._result;;
+*];;
+// Agent should check by this template that initiated action is finished 
+// and that it has result.
+```
+
+<image src="../images/agents/agent_calculate_set_power_specification.png"></image>
+
+## **What are ways of providing the agent's specification?**
+
+The sc-machine API provides two types of functionalities to implement an agent in C++:
+
+a. when the agent's specification is represented in the knowledge base and used by the agent;
+b. when the agent's specification is represented in C++ code and also used by the agent.
+
+In both cases, the agent's specification can be static, dynamic, or semi-dynamic.
+
+1. **Static agent specification** is a specification provided externally in the agent's class (via overriding public getters). This specification isn't saved in the knowledge base because changes in the knowledge base can't automatically alter user-specified code. To provide this type of agent specification, you can use overriding public getters in your agent's class (see documentation below).
+2. **Dynamic agent specification** is a specification provided in the knowledge base or initially in the code but automatically saved into the knowledge base. After that, the first agent changes the specification of the second agent, and the second agent uses the changed specification. To implement this type of specification, use the API of `ScModule` class and the API of `ScAgentBuilder` class (see [**C++ Modules**](modules.md)).
+3. **Semi-dynamic agent specification** can be obtained when the specification is provided in the knowledge base or initially in the code and appended externally (via overriding public getters).
+
+Static agent specification can be useful if you are implementing an agent in C++ for the first time or if you want to minimize the number of searches in the knowledge base. Dynamic agent specification provides the opportunity to analyze and change this specification by other agents. Semi-dynamic specification can be useful if you want to change some parts of this specification but still want the agent's calls to be quick.
+
+---
+
+## **Static agent specification**
+
+Here, the API for implementing an agent with a static specification will be discussed. This is the easiest implementation variant to understand for you. To learn about dynamic agent specification, see [**C++ Modules API**](modules.md).
+
+There are two main classes that you can use to implement an agent: `ScAgent` and `ScActionAgent`.
+
+### **ScAgent**
+
+This class can be used for all classes of agents. The example using this class is represented below.
```

Comment:
what do you mean by `all classes of agents`? Only 2 agent classes were mentioned on this page: platform-independent and platform-dependent. It can confuse users